.github/workflows/ci.yml

```diff
@@ -15,34 +15,63 @@ jobs:
         env:
           GITHUB_CONTEXT: ${{ toJson(github) }}
         run: echo "$GITHUB_CONTEXT"
-  js-lint:
+  js-test-gatsby-build:
+    if: github.ref != 'refs/heads/master'
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - name: Build
-        run: yarn install --frozen-lockfile
-      - name: Lint
-        run: yarn lint
-      - name: Notify deploy failed (#general)
-        if: failure() && github.ref == 'refs/heads/master'
-        uses: pullreminders/slack-action@v1.0.9
-        env:
-          SLACK_BOT_TOKEN: ${{ secrets.DEFUND12_SLACK_DEPLOY_BOT_TOKEN }}
+      - name: Read .nvmrc
+        run: echo ::set-output name=NVMRC::$(cat .nvmrc)
+        id: nvm
+      - name: Use Node.js ${{ steps.nvm.outputs.NVMRC }}
+        uses: actions/setup-node@v1
         with:
-          args: '{\"channel\":\"C014Q9PLEMC\",\"text\":\":warning: Failed to deploy <https://github.com/defund12/defund12.org/commit/${{ github.sha }}|latest commit> on `master` due to issues with `${{ github.job }}` job\"}'
-      - name: Notify deploy failed (#deploys)
-        if: failure() && github.ref == 'refs/heads/master'
-        uses: pullreminders/slack-action@v1.0.9
-        env:
-          SLACK_BOT_TOKEN: ${{ secrets.DEFUND12_SLACK_DEPLOY_BOT_TOKEN }}
+          node-version: "${{ steps.nvm.outputs.NVMRC }}"
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+      - uses: actions/cache@v2
+        id: yarn-cache
         with:
-          args: '{\"channel\":\"C0153GJV1GV\",\"text\":\":warning: Failed to deploy <https://github.com/defund12/defund12.org/commit/${{ github.sha }}|latest commit> on `master` due to issues with `${{ github.job }}` job\"}'
-  js-tests:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+      - name: Install deps
+        run: yarn install --frozen-lockfile --prefer-offline
+      - run: mkdir public
+      - name: build
+        run: yarn build
+      - name: Deploy (push to gh-pages-test branch) # Remove this when merging to master!
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          github_token: ${{ secrets.DEFUND12_GITHUB_TOKEN }}
+          publish_branch: gh-pages-test
```

Comment:
Remove this once we feel good about the outputs of gatsby being pushed to this test branch. Real deploy will happen in `build-and-deploy` step.